app/validators/questions/calculated_question_validator.py

```diff
@@ -19,3 +22,29 @@ def validate_calculations(self):
             for answer_id in calculation["answers_to_calculate"]:
                 if answer_id not in answer_ids:
                     self.add_error(self.ANSWER_NOT_IN_QUESTION, answer_id=answer_id)
+
+    def check_answer_type(self, answer_type, answer_id):
+        if ANSWER_TYPE_TO_JSON_TYPE[answer_type.value] != TYPE_NUMBER:
+            self.add_error(
+                self.ANSWER_TYPE_FOR_CALCULATION_TYPE_INVALID.format(
+                    answer_type=ANSWER_TYPE_TO_JSON_TYPE[answer_type.value],
+                ),
+                identifier=answer_id,
+            )
+
+    def validate_value_source(self):
+        """
+        Validates that source answer or calculated summary is of number type
```

Comment:
We aren't checking calculated summary. Worth adding why.

Reply:
Added a comment and removed the calculated summary mention from here.